adb.js

```diff
@@ -0,0 +1,162 @@
+import usb from 'usb';
+import { USB_VENDOR_IDS, ADB_VALUES, CONNECTION_TYPES } from './lib/constants';
+import { logExceptOnTest } from './lib/helpers';
+import ADBDevice from './lib/adb-device';
+
+const NOT_CONNECTED = 0;
+const WAIT_FOR_AUTH = 1;
+const SEND_PRIVATE_KEY = 2;
+const SEND_PUBLIC_KEY = 3;
+const CONNECTED = 4;
+
+class ADB {
+  constructor (connectionType, device) {
+    if (connectionType !== (CONNECTION_TYPES.USB || CONNECTION_TYPES.TCP)) {
+      throw new Error("Cannot create new ADB device, invalid connection type.");
+    }
+    this.state = NOT_CONNECTED;
+    this.device = new ADBDevice(connectionType, device);
+  }
+
+  // *** STATIC FUNCTIONS ***
+  // return an array of devices that have an ADB interface
+  static findAdbDevices () {
+    logExceptOnTest("Trying to find a usb device");
+    let adbDevices = [];
+    let usbDevices = usb.getDeviceList();
+    let deviceIndex = 0;
```

Comment:
What is `deviceIndex` being used for? It seems to be incremented but never accessed.